### PR TITLE
If B3 header Flag is not set, do not sample the trace.

### PR DESCRIPTION
**Pull request by fedj**
Change-Id: I5c8cd67a78b2baf4482406310cbac21626ed9359

### CODE REVIEW

File: Src/zipkin4net/Src/Trace.cs [hunk 1]
```diff
@@ -150,9 +150,9 @@ private bool ShouldBeRecorded()
                 return (CurrentSpan.Flags & SpanFlags.Sampled) == SpanFlags.Sampled;
             }
 
-            //Backward compatibility mode. If sample flag is not set,
-            //the fact that the trace exists means that it is sampled
-            return true;
+            //If sample flag is not set, do not sample the trace
+            //Avoid issues when B3 header is incomplete
```
Comment (codefromthecrypt) [inline Src/zipkin4net/Src/Trace.cs:154]: just to get this straight..

when X-B3-Sampled is not sent, the receiver should in fact sample the trace ID (deciding to keep it or not)

Is that related to this change?

Comment (fedj) [reply Src/zipkin4net/Src/Trace.cs:154]: Indeed, with some old versions of finagle, the X-B3-Sampled is not sent and thus causing every of these requests to be sampled. This should fix that bug by deciding to not sample it.

Comment (codefromthecrypt) [reply Src/zipkin4net/Src/Trace.cs:154]: but it is valid to not send `X-B3-Sampled`.. the next hop should sample when it isn't sent.

Comment (codefromthecrypt) [reply Src/zipkin4net/Src/Trace.cs:154]: by sample I mean make a decision to keep it or not. In this case, I can see a problem if you are concerned about a buggy client, but non-buggy clients can validly omit the sampled header and such trace IDs should have a fair shot at being chosen.

Comment (fedj) [reply Src/zipkin4net/Src/Trace.cs:154]: oh, you're right! I'll do a fix right now
